modules/ztp-image-based-upgrade-with-backup.adoc

```diff
@@ -0,0 +1,229 @@
+====
+
+.Prerequisites
+
+* You have completed the `Prep` stage.
+* You have installed the OADP Operator.
+* You can an Object storage with your configured credentials.
+// Install OADP and storage configgged: Is it okay to have it in the prereqs? Or is this something that the user would expect to be guided through? With the different storage solutions, we probably can't cover everything.
+
+.Procedure
+
+This example demonstrates how to back up and upgrade a cluster with applications on a persistent volume.
+
+[NOTE]
+====
+The target cluster does not need to be detached from the hub cluster.
+====
+
+. Create the secret with your credentials to your Object storage.
+. Create the configuration for the OADP Operator to access your Object storage instance.
+//TODO Check existing OADP docs for steps, link to them if applicable
+
+. Specify what you want to back up:
+//TODO decide how to present this info. Might be best to create a reference section with all the example CRs.
+
+`backup_acm_klusterlet.yaml`:: This allows you to reattach the cluster to the hub cluster once the upgrade and restore are completed.
+`backup_localvolume.yaml`:: This allows to restore the persistent volume that contains the `index.html` of your application.
+`backup_app.yaml`:: The information that must be backed up is different for each application. This CR specifies what needs to be backed up to recover the applications.
+
+. Define the restore order for the OADP Operator in the `Restore` CR by using the `lca.openshift.io/apply-wave` field:
++
+[source,yaml]
+----
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: acm-klusterlet
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "1"
+spec:
+  backupName:
+    acm-klusterlet
+---
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: apache-app
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "3"
+spec:
+  backupName:
+    apache-app
+---
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: localvolume
+  namespace: openshift-adp
+  labels:
+    velero.io/storage-location: default
+  annotations:
+    lca.openshift.io/apply-wave: "2"
+spec:
+  backupName:
+    localvolume
+----
+
+. Create a `ConfigMap` CR for the backup and restore information that the {lcao} uses during the upgrade.
+
+.. Create a `kustomization.yaml` that appends the information to a `ConfigMap` CR:
++
+[source,yaml]
+----
+configMapGenerator:
+- name: oadp-cm
+  namespace: openshift-adp
+  files:
+  - backup_acm_klusterlet.yaml
+  - backup_localvolume.yaml
+  - backup_app.yaml
+  - restore_acm_klusterlet.yaml
+  - restore_localvolumes.yaml
+  - restore_app.yaml
+----
+
+.. Create the `ConfigMap` CR:
++
+[source,terminal]
+----
+$ kustomize build ./ -o OadpCm.yaml
+----
++
+.Example output
++
+[source,terminal]
+----
+kind: ConfigMap
+metadata:
+  name: oadp-cm-7gtgdhmdc5
```

Comment:
Note: the name of the configMap is autogenerated. I think we can set a "persistent" name instead. However, we should write down that name cause it will be referenced in the IBU oadpContent spec.